### PR TITLE
docs: OEP 63 - TOC Resolution Request

### PR DESCRIPTION
**Pull request by antoviaque**
The TOC discussed during the April 2023 meeting a [series of guidelines for operating and taking decisions](https://discuss.openedx.org/t/2023-04-11-toc-meeting-summary/10019#governance-norms-5). This includes establishing a way for the community to use the TOC to help take decisions, which this OEP implements.

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @antoviaque!  Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.

**Comment by antoviaque** [comment]
@e0d This is a first draft, would you like to do a pass of review?

**Comment by e0d** [comment]
I will take a detailed pass through tomorrow.  Thanks for pulling this together.

**Comment by e0d** [comment]
> It's great to clarify how the TOC can be involved with the community and vice-versa. One suggestion for this OEP: at least in US courts, an appeal is what happens if you don't like a decision that has been made, and you kick it up to a higher authority. That was my first thought when I saw "TOC Appeal."
> 
> Using some of your other words in this draft, maybe we could call it "Request for TOC Decision" ?

I agree with this, I'd even soften it more, "Request of TOC input."

**Comment by itsjeyd** [comment]
label: core contributor

**Comment by antoviaque** [comment]
@nedbat @e0d Good point about the fact that, for a decision to be appealed, it needs to have been taken already :) 

For a better name, "TOC Decision" would work for me. "TOC Input" seem weak - if people want to ask for only "input", they can already do that via the forum for example, or asking TOC members directly. Since we are trying to establish a formal decision process for the TOC based on community input/issues, the name should convey that notion imho. At the same time, since the TOC can also refuse to take the decision (or can have a hard time taking decisions too :) ), I understand the hesitation. So I've looked into alternatives words, and "TOC Arbitration" could be a good one - it conveys the authority and decision-making power of the TOC on the topics submitted, while giving plenty of room to give any type of response the TOC wishes.

**Comment by antoviaque** [comment]
@e0d Btw, since it can be tricky sometimes to get the group to take decisions on a topic, it could be a good occasion to describe a formal mechanism here to help with that? For example, something like that could help ensuring we are able to take decisions, after the topic has been discussed:

* If the proposal contains a decision proposal, it could be submitted to the approval of the TOC during the meeting where it's on the agenda
* Each TOC member can: approve, abstain or object to the proposed decision
* The decision with the most approvals and no objection is considered taken.
* If there are no decision without objection from any TOC member, then the TOC members who have objected to some of the proposed decisions are asked to write an alternative solution.

This way, there would be an easy way to take decisions when there is consensus on what is brought up to the TOC, while leaving the work of drafting a consensual proposal would be left to the community members posting the proposal (if they want a quick decision, they need to make sure there won't be objections to it). On the other side, asking TOC members to write a counter-proposal when they object would ensure people don't object too easily to everything :) and would ensure there is a clear next step when there is no consensus?

**Comment by kdmccormick** [comment]
> For a better name, "TOC Decision" would work for me. "TOC Input" seem weak - if people want to ask for only "input", they can already do that via the forum for example, or asking TOC members directly. Since we are trying to establish a formal decision process for the TOC based on community input/issues, the name should convey that notion imho.

+1

**Review by e0d — CHANGES_REQUESTED** [review]

File: oeps/processes/oep-0063-proc-toc-arbitration.rst
Comment (sarina) [inline oeps/processes/oep-0063-proc-toc-arbitration.rst:76]: Will @e0d always be the TOC chair? I am wary of including info that's likely to change (and forget to be updated) in a foundational document such as an OEP

File: oeps/processes/oep-0063-proc-toc-arbitration.rst
Comment (sarina) [inline oeps/processes/oep-0063-proc-toc-arbitration.rst:79]: This doesn't make sense to me. Isn't the arbitration the TOC decision? So wouldn't the *requester* be able to choose to accept the decision?

File: oeps/processes/oep-0063-proc-toc-arbitration.rst
Comment (sarina) [inline oeps/processes/oep-0063-proc-toc-arbitration.rst:109]: I'm not convinced you need a change history log for changes that occur *within* a single review thread.

**Review by sarina — CHANGES_REQUESTED** [review]

File: oeps/processes/oep-0063-proc-toc-appeal.rst
Comment (antoviaque) [reply oeps/processes/oep-0063-proc-toc-appeal.rst]: @e0d I agree that the original name of appeal did not match well the use case - I replied on this topic to https://github.com/openedx/open-edx-proposals/pull/484#issuecomment-1566819371 and modified the proposal to "arbitration". What do you think?

File: oeps/processes/oep-0063-proc-toc-appeal.rst
Comment (antoviaque) [reply oeps/processes/oep-0063-proc-toc-appeal.rst]: @e0d That would definitely be great to have! And I'm happy to include something about this here. @georgebabey would you like to suggest something inline in this PR? If you have some ideas, I can also give a shot at integrating them in the current document.

File: oeps/processes/oep-0063-proc-toc-appeal.rst
Comment (antoviaque) [reply oeps/processes/oep-0063-proc-toc-appeal.rst]: @e0d Good point - I'm guessing the TOC will still want to have the possibility to do some of the discussion behind closed doors (though there would still be the summary notes published as usual, so it's not fully closed doors), it makes sense for the person who is submitting the request to be able to present it. And if that's a discussion with multiple parties (for example if there is a disagreement on a direction to take), to be fair the presentation would probably have to include the different parties/opinions. 

To also keep it async, maybe it could be under the format of being able to submit presentation videos/materials for the TOC to review async? Any party who would like to could submit a video/presentation async in advance of the meeting. The TOC members, when reviewing the materials in advance, could request some of the parties to join the meeting, to ask questions or discuss.

Let me know what you think? If that works for you, I'll add it to the document.

File: oeps/processes/oep-0063-proc-toc-appeal.rst
Comment (antoviaque) [reply oeps/processes/oep-0063-proc-toc-appeal.rst]: @e0d Great addition - I've included the paragraph at the top: https://github.com/openedx/open-edx-proposals/pull/484/commits/89ce6c1da7cdb6dad40c460efb81d5d5d092e4a1

File: oeps/processes/oep-0063-proc-toc-arbitration.rst
Comment (antoviaque) [reply oeps/processes/oep-0063-proc-toc-arbitration.rst:5]: @sarina My impression was that @e0d didn't like "TOC Decision", and I mentioned some objections to "Request for TOC Input" in https://github.com/openedx/open-edx-proposals/pull/484#issuecomment-1566819371 - so I was trying to find a good alternative. But I would be fine with "TOC Decision" - it has the advantage of being clear and simple.

File: oeps/processes/oep-0063-proc-toc-arbitration.rst
Comment (antoviaque) [reply oeps/processes/oep-0063-proc-toc-arbitration.rst]: Applied in https://github.com/openedx/open-edx-proposals/pull/484/commits/af969657865ac51aa5af842175825ac93da6ebe0

File: oeps/processes/oep-0063-proc-toc-arbitration.rst
Comment (antoviaque) [reply oeps/processes/oep-0063-proc-toc-arbitration.rst]: @sarina Good point, fixed in https://github.com/openedx/open-edx-proposals/pull/484/commits/f04380e25c63eb14da6ae8a89caf2d7c336a37c5

File: oeps/processes/oep-0063-proc-toc-arbitration.rst
Comment (antoviaque) [reply oeps/processes/oep-0063-proc-toc-arbitration.rst]: @sarina Good point. It's useful to be able to give the exact username of the chair for the ping here, but we can replace it with a link to the wiki, which names the TOC chair: https://openedx.atlassian.net/wiki/spaces/COMM/pages/3575939113/Technical+Oversight+Committee+TOC#Details . I have added @e0d 's github and forum usernames there, this way this remains easily available.

File: oeps/processes/oep-0063-proc-toc-arbitration.rst
Comment (antoviaque) [reply oeps/processes/oep-0063-proc-toc-arbitration.rst]: @sarina Good catch - it should be: "If the arbitration _request_ contains a suggested decision, the TOC can choose to accept the suggested decision...". Let me know if that makes more sense? Added in https://github.com/openedx/open-edx-proposals/pull/484/commits/262bdd181b053595f0bd16d43a7ad19a82360940

File: oeps/processes/oep-0063-proc-toc-arbitration.rst
Comment (antoviaque) [reply oeps/processes/oep-0063-proc-toc-arbitration.rst:111]: @sarina What would be the right threshold of changes for adding an history line? You would only add one per pull request that modifies the OEP document?

File: oeps/processes/oep-0063-proc-toc-arbitration.rst
Comment (antoviaque) [reply oeps/processes/oep-0063-proc-toc-arbitration.rst]: @georgebabey That's a good point, it would make sense to make the arbitration request always be a github ticket - besides the split discussion that you mention, it would likely help processing to have a single consistent way to make the request. Maybe the option of the forum post could be replaced by a requirement to notify in the existing public discussion that an arbitration request has been made, and to link to the github ticket representing it?

File: oeps/processes/oep-0063-proc-toc-arbitration.rst
Comment (sarina) [reply oeps/processes/oep-0063-proc-toc-arbitration.rst]: Yes, that clarifies. Thank you!

File: oeps/processes/oep-0063-proc-toc-arbitration.rst
Comment (sarina) [reply oeps/processes/oep-0063-proc-toc-arbitration.rst:5]: @kdmccormick agreed with "TOC Decision" here: https://github.com/openedx/open-edx-proposals/pull/484#issuecomment-1602763927 , and I didn't see Ed oppose it. I support TOC Decision as well. I'm not sure where "Arbitration" came from but in America, Arbitration is often a detestable process (ie in order to receive health care sometimes, you have to agree to arbitration in case something goes wrong, and other sort of ways corporations force you into arbitration)

Comment (antoviaque) [reply oeps/processes/oep-0063-proc-toc-arbitration.rst]: @sarina OK - since I'm also good with "TOC Decision", so I have made the change in https://github.com/openedx/open-edx-proposals/pull/484/commits/55d29d8b0cf87ad570c9f7fad6b0cc7ed29f611d :+1:

File: oeps/processes/oep-0063-proc-toc-appeal.rst
Comment (antoviaque) [reply oeps/processes/oep-0063-proc-toc-appeal.rst]: Actually, based on this follow-up conversation, I've changed it again to @nedbat 's proposal, of "TOC Decision": https://github.com/openedx/open-edx-proposals/pull/484#discussion_r1261461702

File: oeps/processes/oep-0063-proc-toc-arbitration.rst
Comment (antoviaque) [reply oeps/processes/oep-0063-proc-toc-arbitration.rst]: @sarina OK, sounds good - I have removed the additional history in https://github.com/openedx/open-edx-proposals/pull/484/commits/c510773ec784d5adb34268d90de4d0dc210b29f0

File: oeps/processes/oep-0063-proc-toc-decision.rst
Comment (e0d) [inline oeps/processes/oep-0063-proc-toc-decision.rst:5]: This title still doesn't capture the gist of this proposal for me.  Something like "Getting TOC Support," though somewhat informal, is really what we're trying to specify I think.

Comment (antoviaque) [reply oeps/processes/oep-0063-proc-toc-decision.rst:5]: @e0d The lack of formality in "Getting TOC Support" is an issue imho - it is meant to be an important formal action, weakening the title weakens the action, and the expected response. 

What's the part that doesn't match the proposal in "TOC Decision"? The fact that it doesn't contain the part about the community requesting it from the TOC? Maybe "TOC Decision Request" would make this aspect clearer?

**Comment by antoviaque** [comment]
@e0d @itsjeyd What is currently the next step for this OEP? I'm not sure from the current status - it's missing a new review pass from reviewers? @georgebabey @sarina not sure if this was explicit for you. If I'm blocking anything, let me know.

**Comment by e0d** [comment]
> * If the proposal contains a decision proposal, it could be submitted to the approval of the TOC during the meeting where it's on the agenda
> * Each TOC member can: approve, abstain or object to the proposed decision
> * The decision with the most approvals and no objection is considered taken.
> * If there are no decision without objection from any TOC member, then the TOC members who have objected to some of the proposed decisions are asked to write an alternative solution.

Just fully groking this now.  I think that the TOC mechanism for voting is already covered in the charter.  I'm not sure why we would create a different set of rules for this case.  However, voting isn't really covered in the text of the OEP itself, so I don't see this as blocking merging of the OEP.

File: oeps/processes/oep-0063-proc-toc-decision.rst
Comment (e0d) [reply oeps/processes/oep-0063-proc-toc-decision.rst:5]: I suppose I see two issues here:
1. I don't think TOC support should be limited to decisions, there should be avenues for support developing a plan or approach as well, when the scope of the change warrants it.
2. I think an active title would be convey the purpose of the document better.  For me this is still leading with the last resort, TOC breaks the tie because the project is stuck.

File: oeps/processes/oep-0063-proc-toc-appeal.rst
Comment (e0d) [reply oeps/processes/oep-0063-proc-toc-appeal.rst]: This can be resolved when we converge here, https://github.com/openedx/open-edx-proposals/pull/484/files#r1316252934

File: oeps/processes/oep-0063-proc-toc-appeal.rst
Comment (e0d) [reply oeps/processes/oep-0063-proc-toc-appeal.rst]: @georgebabey Any progress on this?

**Comment by itsjeyd** [comment]
@antoviaque I had marked this PR as waiting for review mainly because your latest comments (e.g. [this one](https://github.com/openedx/open-edx-proposals/pull/484#discussion_r1282982983) were waiting for input from your reviewers, but also because it was (and still is) marked as waiting for review from @sarina and @georgebabey.

That said I'm not sure how many approving reviews are needed for an OEP like this one. So perhaps it could be merged once you've had a chance to respond to @e0d's most recent review and he's approved the changes?

File: oeps/processes/oep-0063-proc-toc-arbitration.rst
Comment (antoviaque) [reply oeps/processes/oep-0063-proc-toc-arbitration.rst]: @georgebabey I have implemented this as I was suggesting above - let me know if that works?

File: oeps/processes/oep-0063-proc-toc-decision.rst
Comment (antoviaque) [reply oeps/processes/oep-0063-proc-toc-decision.rst]: @e0d Thanks for the review!

> I don't think TOC support should be limited to decisions, there should be avenues for support developing a plan or approach as well, when the scope of the change warrants it.

I see, it could make sense to use a word with a broader meaning. The difficulty is to find one that remains meaningful. How about "TOC Resolution Request"? "Resolution" is defined as _"a formal expression of the opinion or will of the official body"_, and makes it clear that there will be a concrete and well defined outcome, while being broader than just taking a decision.

> I think an active title would be convey the purpose of the document better. For me this is still leading with the last resort, TOC breaks the tie because the project is stuck.

I've made changes to the beginning of the document, to make it broader than this specific example of the TOC breaking a tie, and more about intervening on the topic. See if that's better, or if you still see elements that are too narrow? https://github.com/openedx/open-edx-proposals/pull/484/commits/db0eadd8ef0a86393a5a7fa06a45488a1edb713e#diff-e1f9259a794a90d6d39a6b1fd5cc2b91258a1138f34516ea6db0f9a1ba99c0c3L39-L48

**Comment by antoviaque** [comment]
@itsjeyd I have just done a pass - sorry for the delay on this, I was off for a few weeks. For the criteria for the merge, @e0d would likely be the one who can answer it.

**Review by sarina — APPROVED** [review]
@antoviaque @itsjeyd I do not need to review further on this. I defer to @e0d 's approval

**Comment by itsjeyd** [comment]
@sarina @antoviaque Sounds good, thanks.

**Comment by itsjeyd** [comment]
Hi @e0d, just checking in to see if you'll be able to get back to this PR in the coming days?

**Comment by itsjeyd** [comment]
Hi @e0d, just checking in to see what the next steps are for getting this PR over the line?

**Comment by itsjeyd** [comment]
@antoviaque @e0d What's the latest status of this PR? Are you still planning on getting it merged?

**Comment by antoviaque** [comment]
@itsjeyd I would definitely still want to get this to the finish line! @e0d Do you want to pick a time to iron this out? https://calendly.com/antoviaque/30min

**Comment by itsjeyd** [comment]
Hi @antoviaque and @e0d, any updates here?

**Review by e0d — APPROVED** [review]
@antoviaque thanks for you patience.  I've taken a final pass through this and think that all of my comments have been addressed.  I think we are good to move forward.

**Comment by georgebabey** [comment]
Also apologies for the delay - I think this is a great starting point and appreciate all the work and revisions @antoviaque! I think we should land this as is, we can always make future improvements based on how people are leveraging the TOC.

**Review by georgebabey — APPROVED** [review]

**Comment by antoviaque** [comment]
@e0d @georgebabey Thank you for the reviews! And for the improvements you have contributed. It's great to see the TOC take this step to invite interactions with the broader community. Curious to see what the community will make of it!

@itsjeyd FYI ^ I don't have merge rights here so someone else will need to merge the OEP. I can announce it on the forums afterwards.

**Pull request merged**
